src/NuGet.Clients/NuGet.Console/Xamls/ConsoleContainer.xaml.cs

```diff
@@ -75,6 +84,27 @@ public void NotifyInitializationCompleted()
 
         public void Dispose()
         {
+            if (!_isTelemetryEmitted)
+            {
+                var telemetryEvent = new TelemetryEvent(PackageManagerConsoleLoadCount, new Dictionary<string, object>
+                                {
+                                    { NuGetPMCLoadCount, _windowLoadCount},
```

Comment:
It's called only when you close VS instance. Hiding or switching to other tabs doesn't cause it dispose. This number does increase more than 1. It increases every time user unhide or switch to from other tabs.

![image](https://user-images.githubusercontent.com/8766776/94703543-cc21ae80-02f3-11eb-9543-b3bcc5fcca63.png)

Reply:
Does our XAML instance getting disposed at VS close, rather than PMC window close, sound by-design, or like a bug?  To me, it sounds like a bug, so I'm worried that relying on this behaviour for telemetry will start producing bad telemetry if the underlying memory leak is fixed (at least this memory leak seems to only create a single instance)

Reply:
I am not sure if it's by-design or bug. Maybe @donnie-msft  can answer this.
But considering cost of creating new object keeping same object until VS close is reasonable use of it.  'Cost' = `memory`+ `time delay` to create that object, it takes noticeable delay even with deferred Powershell load from my experience.
So if we create/destroy every time we open/close this tab the we'll see considerable time delay over time.

Reply:
VS has many tool windows. Imagine each of them using memory, even when the window is closed. This work started because the team that investigates out of memory crashes asked us to stop loading PowerShell when they thought it wasn't needed. I feel it's ironic that you're now saying it's ok for PMC to keep GUI objects in memory even when the window is closed. As for perf, customers aren't opening and closing PMC multiple times per second.

Anyway, I feel like you missed my point. My concern is from a technical correctness point of view. If it's working today as you intended by accident, then if someone in the future fixes this memory leak without knowing our telemetry depends on it, then we might start reporting telemetry differently (multiple times per session), and this could mess up our queries/reports.

Reply:
I agree with Andy that something is wrong with our Dispose timeline. 
If you open PMC, then close it, then open it again, it doesn't initialize. It appears that it keeps the PMC components in memory (PSHost, etc). I don't think this is correct behavior.
Example: **Developer PowerShell** (View -> Terminal)  when closed, then reopened, clearly reloads itself. I'm only guessing, but I believe this tool window is calling dispose as soon as the tool window is closed, not when VS is closed.

Reply:
I am not sure what to do here, it's what it's today.  I can add comment about this concern into code.
There is no other easy way of detecting if PMC is loading automatically when you open VS. 
@donnie-msft  if you know how to detect it then please let me know.
I just discovered this detection work around by luck. I think we better make use of it for now  until there is better way of detecting it.

Reply:
We recently investigated the PM UI not cleaning up the UI components, so doing the equivalent seems reasonable here.

Reply:
Took a quick look at Developer PowerShell, and it appears to be out-of-proc. So nevermind that example.
Have we already concluded that it must remain in memory for the remainder of the VS session? Not sure if there may be other examples for just the XAML disposing.